ocean_provider/validation/RBAC.py

```diff
@@ -77,6 +77,40 @@ def get_additional_dids(self):
             for additional_input in additional_inputs
         ]
 
+    def get_data(self):
+        if "data" not in self.request.keys():
+            raise Exception("Data to encrypt is empty.")
+        if (
+            not self._check_if_asset()
+            and not self._check_if_respects_file_encryption_schema()
+        ):
+            raise Exception("Invalid type of data.")
+
+        return self.request["data"]
+
+    def _check_if_asset(self):
+        data = self.request["data"]
+        if isinstance(data, dict):
+            if data.get("version") is not None:
+                return True
+        elif isinstance(data, str):
+            data_dict = json.loads(data)
```

Comment:
you need to add exception handling for the json.loads, e.g. what if the string is "bla bla"? That can not be loaded into a json dict. I would also change the order like this:
- perform json loading if string. if the json conversion fails, we return False
- the resulting dict (coming from the string loading or directly on the payload) needs the same version check and that's it. No code duplication.

Reply:
What if we can verify in the strigified json if the `version: "4.1.0"` exists? Without converting into a dictionary.

Reply:
It should cover all the cases, but if it is not a dictionary then it is invalid anyway, so I would check both.